#819 Part 1 V2 : Per caller filter hotspots

kgaonkar6 opened the pull request:
<!--Hi there, thanks for your contribution! Please take a moment to fill out this template to facilitate the review of your pull request.-->

### Purpose/implementation Section
In general the question in #819 is to check for hotspots in our data that overlap MSKCC cancer database or other known hotspot sites which could have been missed in consensus calls in `snv-callers` module because the call is found in 1 or 2 out of the 4 callers because of caller limitation. 
In this PR I'm using strelka2,mutect2,vardict and lancet calls and gather calls that overlap MSKCC hotspots and known TERT promoter mutations.

#### What scientific question is your analysis addressing?
Gather calls from each callers and filter for `Amino_Acid_Position` which are hotspots per gene or have overlap with the genomic region.

#### What was your approach?

Filtering for hotspot overlapping mutations
- IMPACT == 'HIGH|MODERATE|MODIFIER' to remove any LOW mutations in the given amino acid position in hotspot database
AND 
- Hugo_Symbol  %in% c(hotspot_database_amino_acid$Hugo_Symbol,hotspot_database_genomic$Hugo_Symbol)
AND 
- ( Amino acid hotspot overlap by matching the `Amino_Acid_position` in MSKCC cancer hotspot database 
OR
- TERT promoter region overlap using a genomic region overlap filtering strategy )

#### What GitHub issue does your pull request address?
#819 


### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

#### Which areas should receive a particularly close look?
**Please note there has been an update**
🚨 🚨 🚨 
I am using awk to filter maf for the genes in hotspot so that R doesn't have to read in the big file into RAM which takes a lot of time and often breaks execution on the docker container these filtered files are now being saved in a tmp directory as suggested in the reviews

I checked the paper for a source code : it's in here https://github.com/taylor-lab/hotspots  and they also get Amino_Acid_Position that I use to check over from maf column `Protein_position` https://github.com/taylor-lab/hotspots/blob/733c727bd4b9f7c1a7f4508b9a467b2f31cacf33/funcs.R#L602 so should be consistent. My only concern is that only 2 Indels are captured as overlapping MSKCC does that sound OK? 

#### Is there anything that you want to discuss further?
Do we think the Amino_Acid_Position + Hugo_Symbol matching between over data and MSKCC database is  sufficient ( since it seems in most cases of subtyping/hotspot are actually amino acid site only) or should I do a liftover from hg19 to hg38 for genomic region overlap?. 


#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes


### Results

#### What types of results are included (e.g., table, figure)?
RDS


#### What is your summary of the results?
subsetted file with hotspot calls per caller ( vardict has multiple RDS files that need to be combined in the next step) 

#### Reproducibility Checklist

<!-- Check all those that apply or remove this section if it is not applicable.-->

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

<!-- Please review https://github.com/AlexsLemonade/OpenPBTA-analysis#documenting-your-analysis -->

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

<!-- IF YOUR PULL REQUEST IS A DATA RELEASE, PLEASE REMOVE THE [HTML COMMENT TAG](https://html.com/tags/comment-tag/) FROM THE SECTION BELOW AND COMPLETE THE CHECKLIST-->

<!-- 
### Data Release Checklist
- [ ] Is the table in doc/data-file-descriptions.md up to date?
- [ ] Is doc/data-format.md up to date?
- [ ] Is doc/release-notes.md up to date?
- [ ] Is download-data.sh up to date?
- [ ] Was download-data.sh tested and did it complete without error?
-->

jashapiro reviewed (commented):
Thanks for this @kgaonkar6! And thank you for being the example review (apologies for  springing that on you...)

I like this approach better than making the database enormous, especially because I don't think you need the power of database for this analysis. As you found, it might be nicer to do this in a language that didn't depend so much on in-memory analysis, but as long as we are sticking with R, I think this solution will work pretty well. 

I have not run this code myself at this point; we will also need to add it CI before merging.

A few thoughts/questions up front:

- What is the next step of this analysis? It seems like the main goal here is to generate the lists of potential overlaps with the hotspots. Is the next step to merge these into a "hotspot consensus" or to leave these separated by caller?
Or will we be creating a consensus + hotspot file as described in #819?

- It looks like the RDS files are still in here, but blank... I expect they should be removed. If there is a final file with the added consensus, that would probably be the only thing that needs to be included at the end of this.

I have a lot of suggestions throughout. Some are bigger than others. Part of the challenge is that there are a lot of things going on! If I were working on it, I might start from the inside out. Pick one caller (the smallest, most likely!) and work with the inner function first (`prepMaf()`, though I suggest renaming). You can simplify that function to be more strict in what arguments it expects (vectors rather than strings or data frames when it makes sense). 

After that you can move up to the outer script, and on to handling the larger files.

kgaonkar6 commented on analyses/hotspots-detection/run_overlaps_hotspot.sh:
```diff
@@ -0,0 +1,50 @@
+
+n=1
+for file in $(ls ../../scratch/pbta-snv-vardict.vep.mafa*); do
+ gzip $file ; 
+ Rscript 00-subset-maf.R --maffile ${file}.gz\
```
I'd started with reading ungzipped subsetted files of vardict but those runs would get killed in `docker exec` so I had them gzipped. I can make smaller subsetted files and only save outputs if they are not empty.

jashapiro replied:
It seems odd that this would make a difference. The memory used should be the same either way.

kgaonkar6 replied:
I kind of went the way of grepping the genes of interest first  with `awk` and then do hotspot filtering with the R script as you had mentioned in a previous script. This was it was much faster and didn't have to save large intermediate files from vardict. Should have done this last week when you suggested it!! 😅

kgaonkar6 commented on analyses/hotspots-detection/00-subset-maf.R:
```diff
@@ -0,0 +1,225 @@
+option_list <- list(
+  make_option(c("-m", "--maffile"),type="character",
+              help="Filtered calls from [mutect2|strelka2|vardict|lancet]"),
+  make_option(c("-c", "--caller"), type="character",
+              help="Caller type [mutect2|strelka2|vardict|lancet]"),
```
ah yes, that's right! I will implement that

kgaonkar6 replied:
I did add a `output_file` param to the 00-subset-maf.R script now.

kgaonkar6 commented on analyses/hotspots-detection/utils/prepMaf.R:
```diff
@@ -0,0 +1,113 @@
+                    gene_table,
+                    hotspot_amino_acid_position_df, 
+                    hotspot_genomic_site_df  ) {
+  if (!missing(gene_table)){
+  # gene list to filter sql lite table
+  gene_list <- gene_table$Hugo_Symbol
+  table <- table %>%
+    dplyr::filter(Hugo_Symbol %in% gene_list)
```
Just a vector would be ok, will update it

kgaonkar6 commented on analyses/hotspots-detection/utils/prepMaf.R:
```diff
@@ -0,0 +1,113 @@
+  
+  if (!missing(variant_classification_values)){
+  # Variant_Classification filtering values
+  variant_classification_values <- unlist(str_split(variant_classification_values,pattern = "[|]"))
```
i had filters for variant_classification column before but not being used in this run, I had kept it to have more filtering options if needed in iterations of the analysis but I think like you suggested in the meeting to remove this if it is not needed anymore.

kgaonkar6 commented on analyses/hotspots-detection/utils/prepMaf.R:
```diff
@@ -0,0 +1,113 @@
+  
+  # filtered calls_base 
+  calls_base <- table %>%
+    as.data.frame() %>%
```
I thought of keeping it table/dataframe as input since I was using the same script for the sql database and directly reading the caller maf files here. I guess now that the sql database PR is closed I can update it to be dataframe input only

kgaonkar6 commented on analyses/hotspots-detection/run_overlaps_hotspot.sh:
```diff
@@ -0,0 +1,50 @@
+for file in $(ls ../../scratch/pbta-snv-vardict.vep.mafa*); do
+ gzip $file ; 
+ Rscript 00-subset-maf.R --maffile ${file}.gz\
+        --caller vardict0${n} \
```
I'll try the ${file##*.} with your suggestion, thanks!

kgaonkar6 replied:
Now that I filter the maf with the genes in the hotspot lists before giving it to the R script I didn't have to susbet the file. So a lot of code is updated in the bash script, hopefully it is easier to follow now.

kgaonkar6 commented on analyses/hotspots-detection/00-subset-maf.R:
```diff
@@ -0,0 +1,225 @@
+  "vcf_pos" = "numeric"
+  )
+
+maffile <- data.table::fread(opt$maffile,
+                             stringsAsFactors = FALSE,
+                             col.names = names(maf_coltypes),
+                             colClasses = maf_coltypes)
```
The subsetted vardict files don't have a header row so I had though of adding the `maf_coltypes` and giving the col.names as inputs while reading the maf files. I believe if a header row is detected it only renames the column names but if no header is detected it uses the `col.names` provided as column names. I can check and get back to this

kgaonkar6 commented:
> Thanks for this @kgaonkar6! And thank you for being the example review (apologies for springing that on you...)
Oh no, I should apologize for putting in a PR so close to the meeting. But just wanted to reduce the number of reviews you would have to do if we were eventually going to use the maf files directly instead of the database. Thanks for the detailed review!

> I have not run this code myself at this point; we will also need to add it CI before merging.
✓

> A few thoughts/questions up front:
> 
> * What is the next step of this analysis? It seems like the main goal here is to generate the lists of potential overlaps with the hotspots. Is the next step to merge these into a "hotspot consensus" or to leave these separated by caller?
>   Or will we be creating a consensus + hotspot file as described in #819?

Right! we will create a consensus+ hotspot file as described in #819

 * It looks like the RDS files are still in here, but blank... I expect they should be removed. If there is a final file with the added consensus, that would probably be the only thing that needs to be included at the end of this.

Ok sounds good, I'll put the intermediate RDS files into `scratch/hotspots-detection`


> I have a lot of suggestions throughout. Some are bigger than others. Part of the challenge is that there are a lot of things going on! If I were working on it, I might start from the inside out. Pick one caller (the smallest, most likely!) and work with the inner function first (`prepMaf()`, though I suggest renaming). You can simplify that function to be more strict in what arguments it expects (vectors rather than strings or data frames when it makes sense).
> 
> After that you can move up to the outer script, and on to handling the larger files.

Sounds like a plan, I can organize the run script to run 1 caller in this PR and then in a subsequent PR re-run the code for other callers.

kgaonkar6 commented on analyses/hotspots-detection/00-subset-maf.R:
```diff
@@ -0,0 +1,225 @@
+# Converting caller ID to call Uppercase for error handling
+caller<-tolower(caller)
+
+maf_coltypes = c(
```
moved to input/maf_colname.RDS to be loaded directly in the script

kgaonkar6 commented on analyses/hotspots-detection/00-subset-maf.R:
```diff
@@ -0,0 +1,225 @@
+  "NCBI_Build"="character",
+  "Chromosome" ="character",
+  "Start_Position" ="numeric",
+  "End_Position" = "numeric",
```
ran styler on both R scripts in the module

kgaonkar6 commented on analyses/hotspots-detection/00-subset-maf.R:
```diff
@@ -0,0 +1,225 @@
+  dplyr::select("Amino_Acid_Position","Hugo_Symbol") %>%
+  dplyr::mutate(hotspot_database="MSKCC") %>%
+  unique() %>%
+  as.data.frame()
```
removed xls and saved the 2 files as tsv in `input/hotspots_database`

kgaonkar6 commented on analyses/hotspots-detection/utils/prepMaf.R:
```diff
@@ -0,0 +1,113 @@
+  
+  if (!missing(impact_values)){
+  # IMPACT filtering values
+  impact_values <- unlist(str_split(impact_values,pattern = "[|]"))
```
variant_classification is not used so removed from script and implemented the vector for impact_values filtering values

kgaonkar6 commented on analyses/hotspots-detection/utils/prepMaf.R:
```diff
@@ -0,0 +1,113 @@
+                                                                 "calls_base_gr",
+                                                                 "genomic_hotspot_gr",
+                                                                 "hotspot_database"))])) %>%
+      dplyr::select(-"Hugo_Symbol.1") %>%
```
hmm I did intend to do everything from `as.data.frame(calls_base_g)` but the first column `calls_base_gr` is a genomic range which actually has all the maf columns  as well, so it names `calls_base_gr.Hugo_Symbol` . I'm now adding subsetbyOverlap where I can just do as.data.frame(calls_base_g)

kgaonkar6 commented:
@jharenza  @jashapiro I was not correctly identifying the columns to match for splice and indels hotspots before so I wanted to point out the problem and solutions for that:

Problem :
- splice sites seem to use the HGVSp_Short ` hotspot_database_2017_snv.tsv` notation in Amino_Acid_Position 
-  it was also pointed out by @migbro and @dmiller15 that the Amino_Acid_Position values for indels in `hotspot_database_2017_indel.tsv ` are in-fact combined region for example : 

Hugo_Symbol | Amino_Acid_Position  | Variant_Amino_Acid
-- | -- | -- 
CDKN2A | 27-42 | V28_E33del:4
CDKN2A | 27-42 |  L32_L37del:3
CDKN2A | 27-42 | A36_N39delinsD:1
CDKN2A | 27-42 | E27del:1
CDKN2A | 27-42 | G35_A36del:1
CDKN2A | 27-42 | N39_N42del:1
CDKN2A | 27-42 | R29_A34del:1

My current solutions are : 
- I've updated the matching for splice sites [here](https://github.com/kgaonkar6/OpenPBTA-analysis/blob/63dbb2d95aab6facab9a2f4bd9322c6549845ddb/analyses/hotspots-detection/utils/filterMaf.R#L36-L39) to match to the HGVSp_Short notation.
In the MSKCC hotspot analysis they also do some analysis to look for neighbors around each splice site if the Reference_Allele is less than 3 bases [here](https://github.com/taylor-lab/hotspots/blob/733c727bd4b9f7c1a7f4508b9a467b2f31cacf33/funcs.R#L625-L639) which is not incorporated in the matching process in this module since it seems the Amino_Acid_Position is assigned the chromosome start position (in hg19). I'm not sure how to handle it any further. 

- for indels I have updated the indels hotspot Amino_Acid_Position to the actual Variant_Amino_Acid positions [here](https://github.com/kgaonkar6/OpenPBTA-analysis/blob/63dbb2d95aab6facab9a2f4bd9322c6549845ddb/analyses/hotspots-detection/00-subset-maf.R#L77-L94)  
since they found these sites are hotspots but it is open for discussion as to if we want to look at any 2 consecutive amino acid change in the bigger hotspot region.

jharenza commented:
@kgaonkar6 this is great. I agree with using two consecutive bases _within_ the hotspot indel regions listed in the INDEL file.

jashapiro commented on analyses/hotspots-detection/utils/filterMaf.R:
```diff
@@ -0,0 +1,138 @@
+      End_Position = as.numeric(End_Position)
+    )
+
+  if (!missing(hotspot_database_2017_snv_df) & !missing(hotspot_database_2017_indel_df)) {
```
Just noticing that if one of these is provided but not the other, this will be skipped entirely, which may not be what you want. This logic appears in a few places, so we would want to fix all of them, or potentially fail early if only one of the two is provided.

kgaonkar6 replied:
Thanks for the note! I added a snippet which will fail early in the script if only 1 of the 2 hotspots database is provided asking to provide the complete database which is snv(+splice) and indel both.

jashapiro reviewed (commented):
Hi @kgaonkar6! I like what you have done here, and I am sure it is running much faster than before!

I made a number of comments throughout, but one thing I want to address is your comment about the indels: https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/956#issuecomment-804152945
I think you moved the logic for that to a different location from the one linked in that comment, and you are now checking for anything that overlaps the indel, correct? I was confused by "2 consecutive amino acid change" part of your comment; is that just because an indel will always be specified with at least two locations? I can certainly imagine a clean indel taking out only one AA. 

In any case, I think the interval logic you used might have missed some overlaps where the end points are not completely within a hotspot. I assume we want to keep those as well, so I proposed an alternative, but you should let me know if that was not what you wanted.

One final note: I will be away for the much of next week, so I won't be able to look at this again for a bit.

dmiller15 commented on analyses/hotspots-detection/utils/filterMaf.R:
```diff
@@ -0,0 +1,138 @@
+                        suffix = c(".calls_base",".hotspot_indel_df")  
+      ) %>% 
+      # keep only sites that where 
+      # calls$start >= hotspot_indel$start and
+      # calls$end <= hotspot_indel$stop
```
In the V2 file they provide all of the calls that were used to define these ranges, and in no instance does INDEL extend beyond the start or stop. If the goal is to label hotspots INDELs as they do in the V2 file, the approach in this comment is appropriate.

jashapiro replied:
I am not sure I understand what you are saying here. We have both the hotspot file and a variety of MAF files that we are looking at. That is to say, we are comparing our indel calls to the ones in the hotspot, so we might well have some that extend outside the defined hotspots.

Either way, the logic I present will work for _any_ overlap (including fully contained ones), so unless we have reasons to exclude overlaps that are not fully enclosed by the defined hotspots, I think it would be prudent to switch.

dmiller15 replied:
I'm talking about the original MAF that was used to construct these hotspot files: http://download.cbioportal.org/cancerhotspots/cancerhotspots.v2.maf.gz. The calls in that were then processed to create the INDEL and SNV inputs that Krutika is using to process what you have here. You can parse ranges for that if you like, but it's easier to see in the INDEL input. Using the example that Krutika gave above:

Hugo_Symbol | Amino_Acid_Position | Variant_Amino_Acid
-- | -- | --
CDKN2A | 27-42 | V28_E33del:4
CDKN2A | 27-42 | L32_L37del:3
CDKN2A | 27-42 | A36_N39delinsD:1
CDKN2A | 27-42 | E27del:1
CDKN2A | 27-42 | G35_A36del:1
CDKN2A | 27-42 | N39_N42del:1
CDKN2A | 27-42 | R29_A34del:1

Here you can see the range for one hotspot indel along with the contributing variants they recognized in their cohort MAF (formatted like `HGVSp:supporting variants count`). So here for the range starting at 27 and ending at 42, in no instance does the actual variant extend beyond the bounds of the range. This observation is true for ALL ranges presented in the INDEL input.

The reason I am providing for exclusion of variants that are not fully enclosed in the range is that data from MSKCC doesn't support INDELs that extend beyond the range. Is there a scientific reason to allow any overlap to count as a hotspot?

jashapiro replied:
> Is there a scientific reason to allow any overlap to count as a hotspot?

I would not necessarily count them "as a hotspot", but if the point of this analysis is to identify mutations in our dataset that might have been missed from earlier consensus, then yes, I would say that we should include anything that overlaps a hotspot. If we had a deletion in our data set from position 26-29 in the above example, I would think we would want to note it, even if it was not exactly one of the MSKCC identified mutations.

kgaonkar6 replied:
Here's an example from the [V2 indel hotspot](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/81dbcd5b71161c225fb226cd86b71281e935dc1e/analyses/hotspots-detection/input/hotspots_database/hotspot_database_2017_indel.tsv) for CDKN2A, where 

Hugo_Symbol | Amino_Acid_Position | Variant_Amino_Acid
-- | -- | --
CDKN2A | 27-42 | V28_E33del:4
CDKN2A | 27-42 | L32_L37del:3
CDKN2A | 27-42 | A36_N39delinsD:1
CDKN2A | 27-42 | E27del:1
CDKN2A | 27-42 | G35_A36del:1
CDKN2A | 27-42 | N39_N42del:1
CDKN2A | 27-42 | R29_A34del:1

- `Amino_Acid_Position` is a region that fully overlaps each hotspot and none of the indels start of end outside the given region so we believe that the hotspot start and end AA positions  should be within the given region to be considered as a hotspot according to MSKCC database . 
- actual `Variant_Amino_Acid` found in samples, denoated as `<Variant change>:<number of samples that have the variant change >`

jharenza replied:
Hi @jashapiro - we had initially thought about identifying novel hotspots, but then brought the scope of this back to only _known_ hotspots, so that is why we had thought to only include INDELS within the already defined hotspot region here. I agree that we would probably want to capture INDELS overlapping a part of this region, but perhaps that is beyond the scope because I would also want to verify those in IGV or molecularly in order to call them high-confidence. Does that sound ok to you? cc @jaclyn-taroni because I think Josh is out this week.

jashapiro replied:
> Hi @jashapiro - we had initially thought about identifying novel hotspots, but then brought the scope of this back to only _known_ hotspots, so that is why we had thought to only include INDELS within the already defined hotspot region here. I agree that we would probably want to capture INDELS overlapping a part of this region, but perhaps that is beyond the scope because I would also want to verify those in IGV or molecularly in order to call them high-confidence. Does that sound ok to you?

I understand the motivation not to add new indels that would require validation, but I don't think the current method accomplishes that, which leads me to think that we have two options:

- If we don't want to add any indels not already called, it seems we should be matching to the `Variant_Amino_Acid` call, and only keeping those precise locations. In my mind, that would be too stringent, as indels are notoriously difficult to call precisely. 

- If we are willing to accept calls that do not match the `Variant_Amino_Acid` locations, but do fall within `Amino_Acid_Position` ranges, then I think we have to also include overlaps that extend beyond that range. 
 
My argument is this: looking at [the table above](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/956#discussion_r602476016), a deletion from position 30-38 would not match any of the variants in MSKCC, but would be caught by being within the 27-42 range. On the other hand, a deletion from 26-34 would fall outside the strict range criteria, but I can see no reason to exclude such a deletion if we were to see it.


I will note that there is a secondary question, which I did not think this PR was to address, which is how we determine which of the hotspot-overlapping calls are high confidence. I would doubt that a call only made by VarDict would be high confidence, even if it happened to appear in a known hotspot. As this PR looks only at callers in isolation, I was expecting that would be a later stage where the calls got merged.

Following on the above example, if there were a sample variant call made by one caller as 26-34del and by another caller as 27-34del, I would hope we would want to include it! If we exclude the former call by a strict overlap at this stage, then we might miss valid calls.

jharenza replied:
Yes, these are good points, especially that we are not strictly matching now and that two callers may call the same INDEL with different coordinates. I am on board now :). How much of an overlap of this region do you think is then sufficient for calling an INDEL?

jaclyn-taroni replied:
> How much of an overlap of this region do you think is then sufficient for calling an INDEL?

This is a discussion point for the later PR. For this one, I think we can plan to make the suggested changes and re-request a review from @jashapiro.

kgaonkar6 commented:
> Hi @kgaonkar6! I like what you have done here, and I am sure it is running much faster than before!

Thanks! :)

> I made a number of comments throughout, but one thing I want to address is your comment about the indels: [#956 (comment)](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/956#issuecomment-804152945)
> I think you moved the logic for that to a different location from the one linked in that comment, and you are now checking for anything that overlaps the indel, correct? I was confused by "2 consecutive amino acid change" part of your comment; is that just because an indel will always be specified with at least two locations? I can certainly imagine a clean indel taking out only one AA.
Right! That was an incorrect statement from my end, there can be only one AA indel so I have not implemented the "2 consecutive amino acid change" in code since that would leave out single AA indels as you mentioned correctly. 

> In any case, I think the interval logic you used might have missed some overlaps where the end points are not completely within a hotspot. I assume we want to keep those as well, so I proposed an alternative, but you should let me know if that was not what you wanted.
Like @dmiller15 mentioned we looked at the variants contributing to that region being called in hotspot in their V2 tsv file and all are within (including start and end) the region. 

> One final note: I will be away for the much of next week, so I won't be able to look at this again for a bit.
👍  no worries!

jharenza commented on analyses/hotspots-detection/00-subset-maf.R:
```diff
@@ -0,0 +1,106 @@
+caller <- tolower(caller)
+output_file <- opt$output_file
+
+maf_coltypes <- readRDS(file.path("input", "maf_coltypes.RDS"))
```
I think she meant "2 consecutive bp change"

kgaonkar6 replied:
I though of adding it as RDS mainly because the read_tsv needs readr::col_[data_types]() specifications. Probably there is a way to have a csv that I don't know yet.

kgaonkar6 commented on analyses/hotspots-detection/utils/filterMaf.R:
```diff
@@ -0,0 +1,138 @@
+#' @param table a maf format dataframe
+#' @param impact_values list of IMPACT [optional] values to filter
+#' @param hotspot_amino_acid_position_df  [optional] dataframe with Hugo_Sybol, Amino_Acid_Position
+#' and hotspot_database columns
```
yes I've updated the description now, thanks for the note!

kgaonkar6 commented on analyses/hotspots-detection/run_overlaps_hotspot.sh:
```diff
@@ -0,0 +1,50 @@
+        --genomic_site_hotspot_file $genomic_site_hotspot_file ;
+ n=$(( $n + 1 ));
+done
+
```
removing temp directory will be included in the next PR where we merge the hotspot filtered per caller calls with (3/3) conensus maf.

kgaonkar6 commented on analyses/hotspots-detection/utils/filterMaf.R:
```diff
@@ -0,0 +1,138 @@
+            gsub("p[.]","",HGVSp_Short))
+    ) %>%
+    mutate(
+      Start_Position = as.numeric(Start_Position),
+      End_Position = as.numeric(End_Position)
```
as.numeric() was needed when I was reading the maf files with fread ( the columns were being read in as character types) but not needed now that I read with read_tsv and provide the col_types directly.

kgaonkar6 commented on analyses/hotspots-detection/00-subset-maf.R:
```diff
@@ -0,0 +1,106 @@
+#
+# Read in MSKCC cancer database 
+hotspot_database_2017_snv <- read_tsv(file.path(mskcc_cancer_hotspot_folder, "hotspot_database_2017_snv.tsv"))
+hotspot_database_2017_indel <- read_tsv(file.path(mskcc_cancer_hotspot_folder, "hotspot_database_2017_indel.tsv")) 
```
added col types here

kgaonkar6 commented on analyses/hotspots-detection/utils/filterMaf.R:
```diff
@@ -0,0 +1,138 @@
+    )
+
+  if (!missing(hotspot_database_2017_snv_df) & !missing(hotspot_database_2017_indel_df)) {
+    # filter by amino acid hotspot_database
```
yes, added comment to include splice site HGVSp_Short matching condition here

kgaonkar6 commented on analyses/hotspots-detection/utils/filterMaf.R:
```diff
@@ -0,0 +1,138 @@
+          grepl("Splice_Site",Variant_Classification) ~
+            # since these sites are intronic sites adding
+            # HGSVp_Short instead of Protein_position
+            gsub("p[.]","",HGVSp_Short))
```
added splice site format in the comment description here

jashapiro commented on analyses/hotspots-detection/utils/filterMaf.R:
```diff
@@ -0,0 +1,141 @@
+                        by = c( "Hugo_Symbol"),
+                        suffix = c(".calls_base",".hotspot_indel_df")  
+      ) %>% 
+      # keep only sites that where 
+      # calls$start >= hotspot_indel$start and
+      # calls$end <= hotspot_indel$stop
```
Updating this comment as it didn't match the code after discussion & updates.
```suggestion
      # keep only sites that overlap hotspot indels with  
      # calls$start <= hotspot_indel$stop and
      # calls$end >= hotspot_indel$start
```

jashapiro reviewed (approved):
Somehow I missed that you had actually made the significant changes that I had requested already! Sorry for the delay in re-reviewing (though I guess with the other changes we are discussing it isn't a major block).

My suggestions here just update a couple of comments.
